src/main/java/com/amuzil/omegasource/magus/radix/RadixTree.java

```diff
@@ -58,7 +63,17 @@ private void terminate() {
     }
 
     public void moveDown(Form executedForm) {
-        if(active.getModifiers().size() > 0) active.getModifiers().forEach(modifier -> modifier.print());
+        //add the last Node to the activation Path and store its ModifierData's
+        if (this.lastActivated != null && active != null) {
+            path.addStep(this.lastActivated, active.getModifiers());
+        }
+        this.lastActivated = executedForm;
+
+        if(active.getModifiers().size() > 0 && owner instanceof ServerPlayer player) {
```

Comment:
Same thing here. I don't know if I commented on it in some of the other classes, but it's worth taking a look at.